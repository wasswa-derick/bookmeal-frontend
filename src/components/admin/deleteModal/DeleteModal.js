import React from "react";
import PropTypes from "prop-types";

const DeleteModal = ({ title, confirmDeletion }) => (
  <div className="modal fade" id="confirmDel">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="mealModalLabel">
            Confirm Deletion
          </h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          <p>Are you sure you want to remove this {title}.</p>
        </div>
        <div className="modal-footer">
          <button
            id="btn-no"
            type="button"
            className="btn btn-secondary"
            data-dismiss="modal"
          >
            No
          </button>
          <button
            onClick={confirmDeletion}
            type="button"
            className="btn-del btn btn-primary"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
);

DeleteModal.propTypes = {
  title: PropTypes.string.isRequired,
  confirmDeletion: PropTypes.func.isRequired
};

export default DeleteModal;
