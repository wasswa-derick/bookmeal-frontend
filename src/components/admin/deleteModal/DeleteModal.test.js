import React from "react";
import { shallow } from "enzyme";
import DeleteModal from "./DeleteModal";

describe("delete modal", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <DeleteModal title="something" confirmDeletion={jest.fn} />
    );
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
