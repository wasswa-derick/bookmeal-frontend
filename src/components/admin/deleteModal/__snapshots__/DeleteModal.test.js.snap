// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete modal should render correctly 1`] = `
<div
  className="modal fade"
  id="confirmDel"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="mealModalLabel"
        >
          Confirm Deletion
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          Are you sure you want to remove this 
          something
          .
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          id="btn-no"
          type="button"
        >
          No
        </button>
        <button
          className="btn-del btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Yes
        </button>
      </div>
    </div>
  </div>
</div>
`;
